src/libs/actions/Report.ts

```diff
@@ -2517,10 +2517,12 @@ function navigateToMostRecentReport(currentReport: OnyxEntry<Report>) {
     const isChatThread = ReportUtils.isChatThread(currentReport);
     if (lastAccessedReportID) {
         const lastAccessedReportRoute = ROUTES.REPORT_WITH_ID.getRoute(lastAccessedReportID ?? '');
+        const isFirstRoute = navigationRef?.current?.getState().index === 1;
         // If it is not a chat thread we should call Navigation.goBack to pop the current route first before navigating to last accessed report.
         if (!isChatThread) {
-            // Fallback to the lastAccessedReportID route, if this is first route in the navigator
-            Navigation.goBack(lastAccessedReportRoute);
+            if (!isFirstRoute) {
```

Comment:
I think we can combine this condition with the above `if (!isChatThread && !isFirstRoute)`